### ISSUE TITLE
Configuration: restructure settings

### ISSUE THREAD

**Issue by ljacqu** [AuthMe/AuthMeReloaded#448]
- Restructure the settings so that the settings are grouped together more clearly and coherently.
- Write a migration so that a player's old config can be "converted" to the new one
- Add an entry to the breaking changes page

**Comment by sgdc3** [comment]
pushed a WIP of the new config

**Comment by ljacqu** [comment]
One wish I would personally have would be to move the mysqlColumnSalt option to the data source section, also the group column. I understand that those are optional but I find it confusing that different column configs are 50 lines apart

**Comment by sgdc3** [comment]
Yeah, i haven't reach the end of the file, in the meantime do what do you want with it. ;)

**Comment by ljacqu** [comment]
Current state: we did some cleanup inside of config but some similar settings are scattered around, e.g. for registration some stuff is under "registration" and some stuff under "security." Also database stuff vs. hooks stuff, and probably many more things. :)
